sudoku.rb

```diff
@@ -0,0 +1,165 @@
+# Sudoku Rules:
+#   Each cell should contain a digit 1-9 such that:
+#     * horizontal row contains each digit exactly once
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+#
+#       [_ _ _  _ _ _  _ _ _]
+#       [1 2 3  4 5 6  7 8 9]
+#       [_ _ _  _ _ _  _ _ _]
+#
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+#     * vertical column contains each digit exactly once
+#       [_ _ _  _ 1 _  _ _ _]
+#       [_ _ _  _ 2 _  _ _ _]
+#       [_ _ _  _ 3 _  _ _ _]
+#
+#       [_ _ _  _ 4 _  _ _ _]
+#       [_ _ _  _ 5 _  _ _ _]
+#       [_ _ _  _ 6 _  _ _ _]
+#
+#       [_ _ _  _ 7 _  _ _ _]
+#       [_ _ _  _ 8 _  _ _ _]
+#       [_ _ _  _ 9 _  _ _ _]
+#     * each 3x3 box contains each digit exactly once
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+#
+#       [_ _ _  1 2 3  _ _ _]
+#       [_ _ _  4 5 6  _ _ _]
+#       [_ _ _  7 8 9  _ _ _]
+#
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+#       [_ _ _  _ _ _  _ _ _]
+
+
+class Sudoku
+
+  #
+  # @param board [Array]
+  #
+  def initialize(board)
+    @board = board
+  end
+
+  #
+  # @return [Boolean] true if board is valid
+  #
+  def solved?
+    #
+    # TODO
+    #
+
+    board = @board
+    row_count = 0
+    valid_count = 0
+    valid = false
+
+    # check if vals in rows are unique
+    uniq_check = Proc.new do |b|
+      b.each do |row|
+        row_count += 1
+        valid_count += 1 if row == row.uniq
+      end
+      valid_count == row_count ? valid = true : valid = false
+    end
+
+    # check if vals in columns are unique
+    columns = board.transpose
+    uniq_check.call(columns)
+
+    # check if each 3x3 box is unique
+    # creates new array from each 3x3 box
+    box1 = board[0][0..2].concat(board[1][0..2], board[2][0..2])
```

Comment:
instead of a explicitly assembling the boxes, any way we can do this programmatically?